src/ai/defaultai.cc

```diff
@@ -3330,18 +3330,34 @@ void DefaultAI::check_flag_distances(const Time& gametime) {
 // Dealing with diplomacy actions
 void DefaultAI::diplomacy_actions(const Time& gametime) {
 
+	// don't do any diplomacy for the first 10 + x minutes to avoid click races for allies
+	if (gametime < Time((10 + RNG::static_rand(10)) * 60 * 1000)) {
+		return;
+	}
+
+	const Widelands::PlayerNumber mypn = player_number();
+	const Widelands::Player* me = game().get_player(mypn);
+
+	// if we are defeated or the last one in a team leave team
+	if (me->team_number() != 0 &&
+	    (player_statistics.members_in_team(me->team_number()) == 1 || me->is_defeated())) {
```

Comment:
well in this case we would need to check for open requests as well and this would mean duplicating most of the code that is not already safeguarded by me->is_defeated() already so I see no benefit here

Reply:
OK.